### ISSUE TITLE
According to the document, it cannot be accessed after being installed and started by pip

### ISSUE THREAD

**Issue by gg-yiwen** [IDEA-Research/deepdataspace#17]
Screenshots of installation startup and results are as follows
<img width="1535" alt="image" src="https://github.com/IDEA-Research/deepdataspace/assets/83198614/5d898763-cd03-4049-9609-29891137f9ea">
<img width="817" alt="image" src="https://github.com/IDEA-Research/deepdataspace/assets/83198614/197e0228-eb4e-468c-8ba4-09e65c5251a3">
![image](https://github.com/IDEA-Research/deepdataspace/assets/83198614/573ee251-9e22-47b2-90ea-1ad14811c18a)

**Comment by imhuwq** [comment]
Can you give us more detailed information about your envrionments?
- Platform: Win/Mac/Linux
- Arch: x86/x86_64/arm

Besides, can you find any django process running with `ps ux | grep django`?

**Comment by gg-yiwen** [comment]
> Can you give us more detailed information about your envrionments?
> 
> * Platform: Win/Mac/Linux
> * Arch: x86/x86_64/arm
> 
> Besides, can you find any django process running with `ps ux | grep django`?

![image](https://github.com/IDEA-Research/deepdataspace/assets/83198614/e3436d91-f2a5-4f96-94b0-19d391f8c2ad)
Python 3.10.10
no such process

**Comment by imhuwq** [comment]
Can you start the DDS in verbose log mode, `dds --quickstart -V`,  and paste the stdout here?

**Comment by gg-yiwen** [comment]
> Can you start the DDS in verbose log mode, `dds --quickstart -V`, and paste the stdout here?

Started DDS[36825] at http://127.0.0.1:8765.
The DDS tool is importing datasets inside dir in the background: /Users/ligaogang/.deepdataspace/dataset-samples.
Explore other useful commands by: ddsop --help.
You can quit the DDS tool with Ctrl+C.

^Z
[1]  + 36825 suspended  dds --quickstart
(dds-test) ➜  ~ dds --quickstart -V
export DDS_PLATFORM=macDDS).
export DDS_HOME_DIR=/Users/ligaogang
export DDS_SHARED_LIB_DIR=/Users/ligaogang/.deepdataspace/lib
export DDS_SHARED_SSL_LIB=/Users/ligaogang/.deepdataspace/lib/libssl.so.1.1
export DDS_SHARED_CRYPTO_LIB=/Users/ligaogang/.deepdataspace/lib/libcrypto.so.1.1
export DDS_SHARED_CURL_LIB=/Users/ligaogang/.deepdataspace/lib/libcurl.so.4
export DDS_CUR_DIR=/Users/ligaogang/Library/Python/3.9/lib/python/site-packages/deepdataspace/services
export DDS_DB_ENGIN=sqlite3
export DDS_DB_HOST=127.0.0.1
export DDS_DB_PORT=3306
export DDS_DB_NAME=deepdataspace
export DDS_DB_USER=dds
export DDS_DB_PASS=dds
export DDS_REDIS_HOST=127.0.0.1
export DDS_REDIS_PORT=9900
export DDS_REDIS_PASS=5Z60UZT41X
export DDS_REDIS_DBNAME=0
export DDS_REDIS_SELF_HOSTED=True
export DDS_REDIS_DIR=/Users/ligaogang/.deepdataspace/redis
export DDS_REDIS_BIN=/Users/ligaogang/.deepdataspace/redis/bin/redis-server
export DDS_REDIS_CONF=/Users/ligaogang/.deepdataspace/redis/redis.conf
export DDS_REDIS_LOG=/Users/ligaogang/.deepdataspace/redis/redis.log
export DDS_REDIS_PID=/Users/ligaogang/.deepdataspace/redis/redis.pid
export DDS_MONGODB_HOST=127.0.0.1
export DDS_MONGODB_PORT=9800
export DDS_MONGODB_USER=ligaogang
export DDS_MONGODB_PASS=KZSR4MMH4U
export DDS_MONGODB_DBNAME=dds
export DDS_MONGODB_SELF_HOSTED=True
export DDS_MONGODB_DIR=/Users/ligaogang/.deepdataspace/mongodb
export DDS_MONGODB_BIN=/Users/ligaogang/.deepdataspace/mongodb/bin/mongod
export DDS_MONGODB_CONF=/Users/ligaogang/.deepdataspace/mongodb/mongo.conf.yaml
export DDS_MONGODB_LOG=/Users/ligaogang/.deepdataspace/mongodb/mongo.log
export DDS_MONGODB_PID=/Users/ligaogang/.deepdataspace/mongodb/mongo.pid
export DDS_MONGODB_DBDIR=/Users/ligaogang/.deepdataspace/mongodb/db
export DDS_DJANGO_HOST=127.0.0.1
export DDS_DJANGO_PORT=8765
export DDS_DJANGO_DIR=/Users/ligaogang/.deepdataspace/django
export DDS_DJANGO_LOG=/Users/ligaogang/.deepdataspace/django/django.log
export DDS_DJANGO_KEY=RXOPI3I7AKRT31BMCGEVPL593EHMVG71
export DDS_DJANGO_KEY_FILE=/Users/ligaogang/.deepdataspace/django/secret.txt
export DJANGO_SETTINGS_MODULE=deepdataspace.server.settings
export DDS_CELERY_DIR=/Users/ligaogang/.deepdataspace/celery
export DDS_CELERY_LOG=/Users/ligaogang/.deepdataspace/celery/celery.log
export DDS_CELERY_WORKER_POOL=solo
export DDS_DATA_DIR=/Users/ligaogang/.deepdataspace/dataset-samples
export DDS_VERBOSE_LOG=True
INFO:dds.utils.os:port 9900 is in use
trying to start redis with command: /Users/ligaogang/.deepdataspace/redis/bin/redis-server redis.conf --dir /Users/ligaogang/.deepdataspace/redis --logfile /Users/ligaogang/.deepdataspace/redis/redis.log --pidfile /Users/ligaogang/.deepdataspace/redis/redis.pid --port 9901
redis is already started, killing it now...
redis is started
export DDS_REDIS_PORT=9901
INFO:dds.utils.os:port 9800 is in use
trying to start mongodb with command: /Users/ligaogang/.deepdataspace/mongodb/bin/mongod -f /Users/ligaogang/.deepdataspace/mongodb/mongo.conf.yaml --dbpath /Users/ligaogang/.deepdataspace/mongodb/db --logpath /Users/ligaogang/.deepdataspace/mongodb/mongo.log --pidfilepath /Users/ligaogang/.deepdataspace/mongodb/mongo.pid --auth --port 9801
mongodb is already started, killing it now...
{"t":{"$date":"2023-06-08T03:04:37.671Z"},"s":"I",  "c":"CONTROL",  "id":20697,   "ctx":"-","msg":"Renamed existing log file","attr":{"oldLogPath":"/Users/ligaogang/.deepdataspace/mongodb/mongo.log","newLogPath":"/Users/ligaogang/.deepdataspace/mongodb/mongo.log.2023-06-08T03-04-37"}}
mongodb is started
export DDS_MONGODB_PORT=9801
trying to start celery with command: /Library/Developer/CommandLineTools/usr/bin/python3 -m celery -A deepdataspace.task:app worker -l info -c 1 --logfile=/Users/ligaogang/.deepdataspace/celery/celery.log


celery@bogon v5.2.7 (dawn-chorus)

macOS-13.0.1-x86_64-i386-64bit 2023-06-08 03:04:41

[config]
.> app:         dds:0x106bf30d0
.> transport:   redis://:**@127.0.0.1:9901/0
.> results:     disabled://
.> concurrency: 1 (prefork)
.> task events: OFF (enable -E to monitor tasks in this worker)

[queues]
.> celery           exchange=celery(direct) key=celery


[tasks]
  . FNFPCalculator
  . RankByFlags
  . deepdataspace.task.import_and_process_data_dir
  . deepdataspace.task.import_and_process_dataset
  . deepdataspace.task.ping.ping

2023-06-08 03:04:41,457 INFO [celery.worker.consumer.connection] Connected to redis://:**@127.0.0.1:9901/0
2023-06-08 03:04:41,463 INFO [celery.worker.consumer.mingle] mingle: searching for neighbors
failed to start celery
Started DDS[37197] at http://127.0.0.1:8765.
The DDS tool is importing datasets inside dir in the background: /Users/ligaogang/.deepdataspace/dataset-samples.
Explore other useful commands by: ddsop --help.
You can quit the DDS tool with Ctrl+C.

2023-06-08 03:04:42,474 INFO [celery.worker.consumer.mingle] mingle: all alone
2023-06-08 03:04:42,492 INFO [celery.apps.worker] celery@bogon ready.
2023-06-08 03:04:42,495 INFO [celery.worker.strategy] Task deepdataspace.task.import_and_process_data_dir[ca8e8fa3-5977-4ab0-9846-505747a45890] received
2023-06-08 03:04:42,497 INFO [celery] import_and_check_data_dir starts, data_dir=/Users/ligaogang/.deepdataspace/dataset-samples
2023-06-08 03:04:42,501 INFO [celery] import_and_process_dataset starts, dataset_dir=/Users/ligaogang/.deepdataspace/dataset-samples/bbox-sample.fasterrcnn.pred, enforce=False
2023-06-08 03:04:46,447 INFO [celery] import_dataset starts, target_path=/Users/ligaogang/.deepdataspace/dataset-samples/bbox-sample.fasterrcnn.pred, enforce=False
2023-06-08 03:04:46,448 INFO [io.importer] import_dataset starts, target_path=/Users/ligaogang/.deepdataspace/dataset-samples/bbox-sample.fasterrcnn.pred, enforce=False
2023-06-08 03:04:46,449 WARNING [io.importer] cannot find any importer for /Users/ligaogang/.deepdataspace/dataset-samples/bbox-sample.fasterrcnn.pred, skip it...
2023-06-08 03:04:46,449 INFO [celery] import_and_process_dataset starts, dataset_dir=/Users/ligaogang/.deepdataspace/dataset-samples/bbox-sample.tsv, enforce=False
2023-06-08 03:04:46,450 INFO [celery] import_dataset starts, target_path=/Users/ligaogang/.deepdataspace/dataset-samples/bbox-sample.tsv, enforce=False
2023-06-08 03:04:46,451 INFO [io.importer] import_dataset starts, target_path=/Users/ligaogang/.deepdataspace/dataset-samples/bbox-sample.tsv, enforce=False
2023-06-08 03:04:46,451 INFO [io.importer] choose_importer_cls: TSVImporter is chosen for target_path /Users/ligaogang/.deepdataspace/dataset-samples/bbox-sample.tsv
2023-06-08 03:04:46,571 WARNING [py.warnings] /Users/ligaogang/Library/Python/3.9/lib/python/site-packages/pymongo/topology.py:177: UserWarning: MongoClient opened before fork. Create MongoClient only after forking. See PyMongo's documentation for details: https://pymongo.readthedocs.io/en/stable/faq.html#is-pymongo-fork-safe
  warnings.warn(

2023-06-08 03:04:46,595 INFO [io.importer] Dataset bbox-sample is already imported, skip.
2023-06-08 03:04:46,596 INFO [celery] import_dataset starts, dataset_dir=/Users/ligaogang/.deepdataspace/dataset-samples/bbox-sample.tsv, enforce=False
2023-06-08 03:04:46,597 INFO [process.processor] process_dataset starts, dataset_dir=/Users/ligaogang/.deepdataspace/dataset-samples/bbox-sample.tsv, enforce=False
2023-06-08 03:04:46,597 INFO [process.processor] FNFPCalculator starts to run, enforce=False
2023-06-08 03:04:46,598 INFO [process.processor] /Users/ligaogang/.deepdataspace/dataset-samples/bbox-sample.tsv is processed before, skip it...
2023-06-08 03:04:46,608 INFO [celery] import_and_process_dataset starts, dataset_dir=/Users/ligaogang/.deepdataspace/dataset-samples/segmentation-sample.tsv, enforce=False
2023-06-08 03:04:46,609 INFO [celery] import_dataset starts, target_path=/Users/ligaogang/.deepdataspace/dataset-samples/segmentation-sample.tsv, enforce=False
2023-06-08 03:04:46,609 INFO [io.importer] import_dataset starts, target_path=/Users/ligaogang/.deepdataspace/dataset-samples/segmentation-sample.tsv, enforce=False
2023-06-08 03:04:46,610 INFO [io.importer] choose_importer_cls: TSVImporter is chosen for target_path /Users/ligaogang/.deepdataspace/dataset-samples/segmentation-sample.tsv
2023-06-08 03:04:46,615 INFO [io.importer] Dataset segmentation-sample is already imported, skip.
2023-06-08 03:04:46,615 INFO [celery] import_dataset starts, dataset_dir=/Users/ligaogang/.deepdataspace/dataset-samples/segmentation-sample.tsv, enforce=False
2023-06-08 03:04:46,616 INFO [process.processor] process_dataset starts, dataset_dir=/Users/ligaogang/.deepdataspace/dataset-samples/segmentation-sample.tsv, enforce=False
2023-06-08 03:04:46,616 INFO [process.processor] FNFPCalculator starts to run, enforce=False
2023-06-08 03:04:46,618 INFO [process.processor] /Users/ligaogang/.deepdataspace/dataset-samples/segmentation-sample.tsv is processed before, skip it...
2023-06-08 03:04:46,618 INFO [celery] import_and_process_dataset starts, dataset_dir=/Users/ligaogang/.deepdataspace/dataset-samples/alphamask-sample.matteformer.pred, enforce=False
2023-06-08 03:04:46,650 INFO [celery] import_dataset starts, target_path=/Users/ligaogang/.deepdataspace/dataset-samples/alphamask-sample.matteformer.pred, enforce=False
2023-06-08 03:04:46,651 INFO [io.importer] import_dataset starts, target_path=/Users/ligaogang/.deepdataspace/dataset-samples/alphamask-sample.matteformer.pred, enforce=False
2023-06-08 03:04:46,652 WARNING [io.importer] cannot find any importer for /Users/ligaogang/.deepdataspace/dataset-samples/alphamask-sample.matteformer.pred, skip it...
2023-06-08 03:04:46,652 INFO [celery] import_and_process_dataset starts, dataset_dir=/Users/ligaogang/.deepdataspace/dataset-samples/annotation-sample.tsv, enforce=False
2023-06-08 03:04:46,653 INFO [celery] import_dataset starts, target_path=/Users/ligaogang/.deepdataspace/dataset-samples/annotation-sample.tsv, enforce=False
2023-06-08 03:04:46,653 INFO [io.importer] import_dataset starts, target_path=/Users/ligaogang/.deepdataspace/dataset-samples/annotation-sample.tsv, enforce=False
2023-06-08 03:04:46,654 INFO [io.importer] choose_importer_cls: TSVImporter is chosen for target_path /Users/ligaogang/.deepdataspace/dataset-samples/annotation-sample.tsv
2023-06-08 03:04:46,657 INFO [io.importer] Dataset annotation-sample is already imported, skip.
2023-06-08 03:04:46,657 INFO [celery] import_dataset starts, dataset_dir=/Users/ligaogang/.deepdataspace/dataset-samples/annotation-sample.tsv, enforce=False
2023-06-08 03:04:46,658 INFO [process.processor] process_dataset starts, dataset_dir=/Users/ligaogang/.deepdataspace/dataset-samples/annotation-sample.tsv, enforce=False
2023-06-08 03:04:46,658 INFO [process.processor] FNFPCalculator starts to run, enforce=False
2023-06-08 03:04:46,660 INFO [process.processor] /Users/ligaogang/.deepdataspace/dataset-samples/annotation-sample.tsv is processed before, skip it...
2023-06-08 03:04:46,660 INFO [celery] import_and_process_dataset starts, dataset_dir=/Users/ligaogang/.deepdataspace/dataset-samples/segmentation-sample.maskrcnn.pred, enforce=False
2023-06-08 03:04:46,661 INFO [celery] import_dataset starts, target_path=/Users/ligaogang/.deepdataspace/dataset-samples/segmentation-sample.maskrcnn.pred, enforce=False
2023-06-08 03:04:46,661 INFO [io.importer] import_dataset starts, target_path=/Users/ligaogang/.deepdataspace/dataset-samples/segmentation-sample.maskrcnn.pred, enforce=False
2023-06-08 03:04:46,679 WARNING [io.importer] cannot find any importer for /Users/ligaogang/.deepdataspace/dataset-samples/segmentation-sample.maskrcnn.pred, skip it...
2023-06-08 03:04:46,680 INFO [celery] import_and_process_dataset starts, dataset_dir=/Users/ligaogang/.deepdataspace/dataset-samples/alphamask-sample.tsv, enforce=False
2023-06-08 03:04:46,680 INFO [celery] import_dataset starts, target_path=/Users/ligaogang/.deepdataspace/dataset-samples/alphamask-sample.tsv, enforce=False
2023-06-08 03:04:46,681 INFO [io.importer] import_dataset starts, target_path=/Users/ligaogang/.deepdataspace/dataset-samples/alphamask-sample.tsv, enforce=False
2023-06-08 03:04:46,681 INFO [io.importer] choose_importer_cls: TSVImporter is chosen for target_path /Users/ligaogang/.deepdataspace/dataset-samples/alphamask-sample.tsv
2023-06-08 03:04:46,684 INFO [io.importer] Dataset alphamask-sample is already imported, skip.
2023-06-08 03:04:46,684 INFO [celery] import_dataset starts, dataset_dir=/Users/ligaogang/.deepdataspace/dataset-samples/alphamask-sample.tsv, enforce=False
2023-06-08 03:04:46,685 INFO [process.processor] process_dataset starts, dataset_dir=/Users/ligaogang/.deepdataspace/dataset-samples/alphamask-sample.tsv, enforce=False
2023-06-08 03:04:46,685 INFO [process.processor] FNFPCalculator starts to run, enforce=False
2023-06-08 03:04:46,686 INFO [process.processor] /Users/ligaogang/.deepdataspace/dataset-samples/alphamask-sample.tsv is processed before, skip it...
2023-06-08 03:04:46,687 INFO [celery.app.trace] Task deepdataspace.task.import_and_process_data_dir[ca8e8fa3-5977-4ab0-9846-505747a45890] succeeded in 4.191095646000001s: None

**Comment by imhuwq** [comment]
Is this the full log outputs?
I can't find any log like `trying to start django ...`, which should be printed just before starting django process.

**Comment by gg-yiwen** [comment]
> Is this the full log outputs? I can't find any log like `trying to start django ...`, which should be printed just before starting django process.
It `s complete
![image](https://github.com/IDEA-Research/deepdataspace/assets/83198614/ad238e89-446d-4b04-a43c-e6780e40e277)

**Comment by imhuwq** [comment]
Can you screenshot down the log outputs just after you start the DDS tool?
By the way, We are working on a docker image which might ease the starting pain.

**Comment by gg-yiwen** [comment]
> Can you screenshot down the log outputs just after you start the DDS tool? By the way, We are working on a docker image which might ease the starting pain.

![image](https://github.com/IDEA-Research/deepdataspace/assets/83198614/c8d304d4-98bc-4d71-b4cb-18592ec7d2f5)
![image](https://github.com/IDEA-Research/deepdataspace/assets/83198614/31c78414-fd76-4457-b55c-1fbf97e51526)

**Comment by imhuwq** [comment]
Can you check the installation of Django package?
`pip list | grep Django` and `which django-admin`

**Comment by gg-yiwen** [comment]
> Can you check the installation of Django package? `pip list | grep Django` and `which django-admin`

![image](https://github.com/IDEA-Research/deepdataspace/assets/83198614/08419d3c-8edb-44d6-91e6-8fbb14ae1b90)

**Comment by imhuwq** [comment]
I have reproduced your issue with Python 3.9, working on it.

**Comment by imhuwq** [comment]
I have fixed this problem on branch [bugfix/find_dds_subprocess](https://github.com/IDEA-Research/deepdataspace/tree/bugfix/find_dds_subprocess).
An official release including this fix will be released tomorrow.

Or you can re-install DDS from source code of this branch:

```shell
git clone https://github.com/IDEA-Research/deepdataspace.git
git checkout -b bugfix/find_dds_subprocess origin/bugfix/find_dds_subprocess

# prepare the node environment（if you haven't installed the Pnpm and Node environment yet）
curl -fsSL https://get.pnpm.io/install.sh | sh -
pnpm env use --global lts

# compile frontend files
pnpm i
pnpm run build:app

# copy frontend files to python package dir
rm -rf deepdataspace/server/static/*
cp -R packages/app/dist/* deepdataspace/server/static/
cp deepdataspace/server/static/index.html deepdataspace/server/templates/

# install the package
python3 -m pip install pip --upgrade
python3 -m pip install -r requirements.txt
python3 setup.py install
```

**Comment by imhuwq** [comment]
#21

**Comment by gg-yiwen** [comment]
> 

I try to install from source code, but the next step is always unsuccessful. Is there any good way?
`curl -fsSL https://get.pnpm.io/install.sh | sh -`

**Comment by imhuwq** [comment]
We just released v0.2.0 on PyPi, which fixed this problem.

```shell
pip install deepdataspace -U
```
